linux/ibus-keyman/src/keyman-system-service-client.c

```diff
@@ -0,0 +1,90 @@
+#include <systemd/sd-bus.h>
+#include "keyman-system-service-client.h"
+
+extern gboolean testing;
+
+#define KEYMAN_BUS_NAME       "com.keyman.SystemService1"
+#define KEYMAN_INTERFACE_NAME "com.keyman.SystemService1.System"
+#define KEYMAN_OBJECT_PATH    "/com/keyman/SystemService1/System"
+
+static void
+finish(sd_bus_error* error, sd_bus_message* msg, sd_bus * bus) {
+  if (error) { sd_bus_error_free(error); }
+  if (msg)   { sd_bus_message_unref(msg); }
+  if (bus)   { sd_bus_unref(bus); }
+}
```

Comment:
I think I'd prefer to see the `goto cleanup` pattern which I've noted in another comment -- it's less vulnerable to copypasta errors.